File: TripNest_Backend/src/main/java/com/tripnest/backend/service/ActivityService.java
package com.tripnest.backend.service;

import java.util.List;

import com.tripnest.backend.common.ApiResponse;
import com.tripnest.backend.dto.CreateActivityRequest;
import com.tripnest.backend.dto.UpdateActivityRequest;
import com.tripnest.backend.dto.response.ActivityResponse;

import jakarta.validation.Valid;

public interface ActivityService {

    ApiResponse<ActivityResponse> createActivity(Long tripId, Integer dayNumber,
            CreateActivityRequest request);

    ApiResponse<List<ActivityResponse>> getActivities(
            Long itineraryId);

    ApiResponse<ActivityResponse> updateActivity(
            Long activityId,
            @Valid UpdateActivityRequest request);

    ApiResponse<String> deleteActivity(
            Long activityId);

}

File: TripNest_Frontend/src/services/activityService.js
import api from "./api";

export const createActivity = (tripId, dayNumber, activity) => {
    return api.post(
        `/trips/${tripId}/days/${dayNumber}/activities`,
        activity
    );
};

export const updateActivity = (activityId, activity) => {
    return api.put(
        `/trips/activities/${activityId}`,
        activity
    );
};

export const deleteActivity = (activityId) => {
    return api.delete(
        `/trips/activities/${activityId}`
    );
};